scripts/fetch_vault_credentials.sh

```diff
@@ -1,4 +1,12 @@
 #!/usr/bin/env bash
 
 SCRIPTPATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-eval "$($SCRIPTPATH/fetch_vault_credentials.py)"
+
+PYTHON_SCRIPT_OUTPUT=$($SCRIPTPATH/fetch_vault_credentials.py)
+PYTHON_SCRIPT_EXIT_CODE=$?
+
+if [ ${PYTHON_SCRIPT_EXIT_CODE} == 0 ]; then
+    eval "${PYTHON_SCRIPT_OUTPUT}"
+else
+    exit ${PYTHON_SCRIPT_EXIT_CODE}
+fi
```

Comment:
```suggestion
set -euo pipefail
PYTHON_SCRIPT_OUTPUT="$($SCRIPTPATH/fetch_vault_credentials.py)"
eval "$PYTHON_SCRIPT_OUTPUT"
```
I think we can do this more safely with `set -euo pipefail` (I think the set -e is the only important bit but the others are some extra safety. This avoids us having to check error codes ourselves.

This script never echos:
```
#!/usr/bin/env bash
set -euo pipefail

foo="$(exit 1)"
echo "Never got here"
```